SnapSpire/Features/Discover/View/SectionListView.swift

```diff
@@ -0,0 +1,44 @@
+import SwiftUI
+
+struct SectionListView: View {
+    var title: String
+    var action: () -> Void
+    var imgWidth: CGFloat
+    var imgHeight: CGFloat
+    var type: ImageType?
+
+    var items: [SectionItem] {
+        SectionItem.topics.filter { $0.type == type }
+    }
+
+    let columns = [GridItem(.flexible()), GridItem(.flexible())]
+
+    var body: some View {
+        VStack {
+            if !items.isEmpty {
+                HeaderView(sectionName: title, sectionAction: { action() })
+            }
+
+            ScrollView(.horizontal, showsIndicators: false) {
+                HStack {
+                    ForEach(items, id: \.self) { item in
+                        NavigationLink(value: item) {
+                            CardView(image: item.image, name: item.name, imageWidth: imgWidth, imageHeight: imgHeight)
+                        }
+                    }
+                }
+            }
+            .navigationDestination(for: SectionItem.self) { item in
```

Comment:
if we instantiate 3 SectionListView, we will have 3 navigation destination modifiers and this will cause problems, we should have only 1 navigation destination for a specific type like SectionItem.self